ocaml/idl/ocaml_backend/genOCaml.ml

```diff
@@ -172,6 +179,10 @@ let gen_of_xmlrpc api tys = block
       ["let methodResponse = From.methodResponse"];
       ["let set f (xml: XMLRPC.xmlrpc) =";
        "  From.array f xml"];
+      ["let list_to_option = function";
+       "| [] -> None";
+       "| [x] -> Some x";
+       "| _ -> failwith \"0 or 1 list elements expected for option type\""];
```

Comment:
Add a `log_backtrace` and raise a proper error? Not that is super important, this is going to disappear after https://github.com/xapi-project/xen-api/pull/3490 lands

Reply:
Indeed, good point, but I think I'll just take your changes into our branch once it's ready.